internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -98,77 +98,98 @@ func (md *machineDeployment) deployMachinesApp(ctx context.Context) error {
 	processGroupMachineDiff := md.resolveProcessGroupChanges()
 	md.warnAboutProcessGroupChanges(ctx, processGroupMachineDiff)
 
-	if len(processGroupMachineDiff.machinesToRemove) > 0 {
-		// Destroy machines that don't fit the current process groups
-		if err := md.machineSet.RemoveMachines(ctx, processGroupMachineDiff.machinesToRemove); err != nil {
-			return err
+	// TODO: validate _somewhere_ no mounts/volumes are used.
```

Comment:
aside from this, looks good!
nice drive-by cleanup as well, some of the deploy code definitely needed a little love

Reply:
@dangra is there a central place where do validate the fly.toml? My thought is if we can rely on the config being valid, I don't need to care about it at this point.

Reply:
> I thought we were skipping canary and falling back to Rolling in that case?

Is that what we want to do? Or rather, is that what we do now?

Thanks for the specific code points! I'll work on getting the validation in place.

Reply:
> Is that what we want to do? Or rather, is that what we do now?

You are right, failing is what we do now ([source](https://github.com/superfly/web/blob/a1f46ab04f52962e0904e6860167003f7bd86470/app/models/nomad/mutations/update_strategy.rb#L12-L17))

Reply:
Went with https://github.com/superfly/flyctl/pull/2279/commits/e49507c1650c9a3e6d21299c0415a0796bff9c10 but not sure if there's any side effects with setting volumes earlier.

Reply:
I think we have to revert the setVolumes() change and look for len(appconfig.Mounts) instead.

Reply:
btw, the appconfig validation is the right thing to do IMO. Checking it here is a bit late.

Reply:
Yea... I initially started adding code to `cfg.validateDeploySection` but `Validate` doesn't take the `MachineDeploymentArgs` deployment as a parameter and it's invoked from several different places which wouldn't have a variable to send there. So my compromise was to do the validation later after creating the `machineDeployment` object which seems to have its own set of validations in the various `setXXX` calls.

Reply:
I see. I can redo it later.  

To me, "[determineAppConfigForMachines](https://github.com/superfly/flyctl/blob/master/internal/command/deploy/machines.go#L83)" is where md.appConfig.Deploy.Strategy should be set if passed as argument. Pretty much the same we do for other arguments like primaryRegion and envvars. And below that, within same function NewMachineDeployment, it runs [validations](https://github.com/superfly/flyctl/blob/master/internal/command/deploy/machines.go#L87).